src/Microsoft.Health.Fhir.Core/Features/Operations/Reindex/Models/ReindexJobRecord.cs

```diff
@@ -82,6 +86,19 @@ protected ReindexJobRecord()
         [JsonProperty(JobRecordProperties.MaximumNumberOfResourcesPerQuery)]
         public uint MaximumNumberOfResourcesPerQuery { get; private set; }
 
+        /// <summary>
+        /// Controls the time between queries of resources to be reindexed
+        /// </summary>
+        [JsonProperty(JobRecordProperties.QueryDelayIntervalInMilliseconds)]
+        public int QueryDelayIntervalInMilliseconds { get; set; }
```

Comment:
>QueryDelayIntervalInMilliseconds [](start = 19, length = 32)

I probably wrong, but it looks for me like it's constant thing. Is it suppose to change at some point or it just 500 all the time.

Reply:
This can be set in the appsettings, and also by the input parameters when you submit a job.

Reply:
Is that thing exist only in scope of throtolling? Is it actually necessary to expose to user? How they suppose to pick that value? With usage percentage you can somewhat estimate load on your server and give it half of capacity of 75% or something like that which is still not the best thing from user perspective because you can encounter spikes in load, and optimally reindex job should be treated as second priority queries.

But that query delay inteval is confusing, it basically exposes implementation details on user shoulders.

---
In reply to: [598963796](https://github.com/microsoft/fhir-server/pull/1750#discussion_r598963796) [](ancestors = 598963796)